hosted-ce-setup.sh

```diff
@@ -13,18 +15,38 @@ if [[ $? -eq 0 ]]; then
   cp /tmp/99-local.ini /etc/osg/config.d/99-local.ini
 fi
 
-# need to programmatically get users
-for user in $(echo $CE_USERS | tr ',' ' '); do
-  echo "Adding user $user"
-  adduser $user 
-  mkdir -p /home/$user/.ssh
-  chown $user: /home/$user/.ssh
-  ssh-keyscan -H $(echo $ENDPOINT | cut -d'@' -f2) >> /home/$user/.ssh/known_hosts
-done
-
-echo "Keyscanning.."
-mkdir -p ~/.ssh
-ssh-keyscan -H $(echo $ENDPOINT | cut -d'@' -f2) >> ~/.ssh/known_hosts
+setup_ssh_config () {
+  echo "Adding user ${ruser}"
+  ssh_dir="/home/${ruser}/.ssh"
+  if ! getent passwd "${ruser}" > /dev/null 2>&1; then
+     # setup user and SSH dir
+     adduser --base-dir /home/ "${ruser}"
+     mkdir -p $ssh_dir
+     chown "${ruser}": $ssh_dir
+     chmod 700 $ssh_dir
+
+     # copy Bosco key
+     ssh_key=$ssh_dir/bosco.key
+     cp /etc/osg/bosco.key $ssh_key
+     chmod 600 $ssh_key
+     chown "${ruser}": $ssh_key
+  fi
+
+  # setup known hosts
+  ssh-keyscan -H "${rhost}" >> $ssh_dir/known_hosts
+
+  # add host SSH config
+  ssh_config=$ssh_dir/config
+  if ! grep -q "^Host ${rhost}$ $ssh_config"; then
```

Comment:
```suggestion
  if ! grep -q "^Host ${rhost}$" $ssh_config; then
```

Reply:
I think the original placement of the end-doublequote treated the ssh config filename as part of the regular expression.

Reply:
Good catch!